msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -90,8 +101,50 @@ public Integer getNumberOfDataPoints() {
 
 	@Override
 	public double[] getMzValues() {
```

Comment:
The getMzValues() method should not be parsing any XML - it should only obtain the Base64-encoded data from the correct position in the memory-mapped mzML file, and decode it.